src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,316 @@
+package duke.command;
+
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.DukeException;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Class that parses the user commands
+ */
+public class Parser {
+
+    /**
+     * Parses the given command of the user
+     * @param command Command given by the user
+     * @param taskList TaskList object containing the existing tasks
+     * @return The reply to the given command
+     * @throws DukeException Error message to indicate invalid command
+     */
+    public static String parseCommand(String command, TaskList taskList) throws DukeException {
+        String[] tokens = command.split(" ", 2);
+        String action = tokens[0].trim().toLowerCase();
+
+        switch (action) {
+            case "list":
+                return generateListResponse(taskList);
+            case "delete":
+                return generateDeleteResponse(tokens, taskList);
+            case "mark":
+                return generateMarkResponse(tokens, taskList);
+            case "unmark":
+                return generateUnmarkResponse(tokens, taskList);
+            case "todo":
+                return generateTodoResponse(tokens, taskList);
+            case "deadline":
+                return generateDeadlineResponse(tokens, taskList);
+            case "event":
+                return generateEventResponse(tokens, taskList);
+            case "find":
+                return generateFindResponse(tokens, taskList);
+            default:
+                throw new DukeException("Sorry, I don't understand what that means.");
+        }
```

Comment:
Perhaps your indentation is wrong according to the coding standards of this module.